docs/src/main/asciidoc/microprofile-graphql.adoc

```diff
@@ -0,0 +1,416 @@
+
+The solution is located in the `microprofile-graphql-quickstart` {quickstarts-tree-url}/microprofile-graphql-quickstart[directory].
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=microprofile-graphql-quickstart \
+    -DclassName="org.acme.microprofile.graphql.FilmResource" \
+    -Dextensions="smallrye-graphql"
+cd microprofile-graphql-quickstart
+----
+
+This command generates a Maven project, importing the `smallrye-graphql` extension which is an implementation of the MicroProfile GraphQL specification used in Quarkus.
+
+== Preparing an Application: GraphQL API
+
+In this section we will start creating the GraphQL API
+
+First, create a the following entities representing a hero from a galaxy far far away:
+
+[source,java]
+----
+package org.acme.microprofile.graphql;
+
+public class Film {
+    public String title;
+    public String episodeID;
+    public String openingCrawl;
+    public String director;
+    public String producers;
+    public LocalDate releaseDate;
+}
+
+public class Hero {
+
+    public String name;
+    public Double height;
+    public Integer mass;
+    public Boolean isDarkSide;
+    public LightSaber lightSaber;
+
+}
+
+enum LightSaber {
+    RED, BLUE, GREEN
+}
+----
+
+The classes we have just created describes the GraphQL schema which is a
+set of possible data (objects, fields, relationships) that a client can access.
+
+Let's continue with an example CDI bean, that would work as a repository.
+
+[source,java]
+
+----
+public class GalaxyService {
+    //..
+}
+----
+
+Now lets use create our first *Graphql* API:
+
+Edit the `org.acme.microprofile.graphql.FilmResource` class as follows:
+
+[source,java]
+----
+@GraphQLApi // <1>
+public class FilmResource {
+
+    @Inject
+    GalaxyService service;
+
+    @Query("allFilms") // <2>
+    @Description("Get all Films from a galaxy far far away") // <3>
+    public List<Film> getAllFilms() {
+        return service.getAllFilms();
+    }
+}
+----
+
+<1> `@GraphQLApi` annotation indicates that the CDI will be a GraphQL endpoint
+<2> `@Query` annotation named `allFilms` defines that the this method will a queryable
+<3> Documentation of the queryable method
+
+TIP:  The value of the `@Query` annotation is optional and would implicitly
+be defaulted to the method name if excluded.
+
+
+This way we have created our first queryable API which we will later expand.
+
+== Launch
+
+Launch the quarkus app:
+[source, bash]
+----
+./mvnw compile quarkus:dev
+----
+
+== Introspect
+
+The full schema of the `graphQL` API can be retrieved by calling the following:
+
+----
+curl http://localhost:8080/graphql/schema.graphql
+----
+
+The server will return the complete schema of the `graphQL` API.
+
```

Comment:
You can maybe mention that in dev and test (and by configuration) GraphiUI will be available in http://localhost:8080/graphql-ui

Reply:
![image](https://user-images.githubusercontent.com/6836179/81271356-43f0f300-904c-11ea-8e90-697ac6750cc2.png)

Reply:
Okey, sounds like a great idea 👍

Reply:
I mentioned Graphiql in the documentation. Great feature 👍